src/router/components/RouterWrapper/RouterWrapper.tsx

```diff
@@ -0,0 +1,24 @@
+import type { PropsWithChildren } from "react";
+import React from "react";
+import { Navigate, Route, Routes } from "react-router-dom";
+
+import { MainLayout } from "~/router";
+import { useAuthStore } from "~/stores";
+
+type Props = PropsWithChildren & {
+  guest?: boolean;
+};
+
+export const RouterWrapper = ({ guest = false, children }: Props) => {
+  const isAuthenticated = useAuthStore((state) => state.token);
+
+  if (!guest && !isAuthenticated) {
```

Comment:
What's the difference between `guest` and `isAuthenticated`? Why do we need both?

Reply:
What about not showing those routes to authenticated users ? 
Why should an authenticated user visit the register page?
This for example is an extract of a project:

```
  {
    element: <AuthenticatedRoute roles={[ROLES.admin]} />,
    children: [
      {
        path: PATHS.externalApis,
        element: <ExternalApiSettingsScreen />,
      },
      {
        path: `${PATHS.externalApis}/:id`,
        element: <ExternalApiDetailsScreen />,
      },
      {
        path: PATHS.reports,
        element: <ReportsScreen />,
      },
    ],
  },
  {
    element: <UnauthenticatedRoute />,
    children: [
      {
        path: '/',
        element: <Navigate to={PATHS.account.login} replace />,
      },
      {
        path: PATHS.account.register,
        element: <RegisterScreen />,
      },
      {
        path: PATHS.account.login,
        element: <LoginScreen />,
      },
      {
        path: PATHS.account.forgotPassword,
        element: <ForgotPasswordScreen />,
      },
    ],
  },
  { path: '*', element: <NotFoundScreen /> },
 ```

Reply:
I like the "Public" Idea from Ferna, I see some cases in where that can be useful, maybe not for a register, but maybe for some info pages or things like that

Reply:
I support the idea of having private / public routes.